django/text_extractor/utils.py

```diff
@@ -131,11 +128,40 @@ def resize_image_to_a4(img, dpi=150):
     return background
 
 
+def resize_to_azure_images(image):
+    width, height = image.size
+    if width < 50 or height < 50:
+        # Resize to at least 50 pixels
+        if width <= height:
+            new_width = 50
+            new_height = int(height * (50 / width))
+        else:
+            new_height = 50
+            new_width = int(width * (50 / height))
+    elif width > 10000 or height > 10000:
+        # Resize to max 10000 pixels
+        if width >= height:
+            new_width = 10000
+            new_height = int(height * (10000 / width))
+        else:
+            new_height = 10000
+            new_width = int(width * (10000 / height))
+    else:
+        new_width, new_height = width, height
+    # Edge case: insanely wide or tall images. Don't maintain proportions.
+    new_width = max(50, min(new_width, 10000))
+    new_height = max(50, min(new_height, 10000))
+
+    # Resize the image
+    image = image.resize((new_width, new_height), Resampling.LANCZOS)
+    return image
+
+
```

Comment:
had to take part of the librarian's method because I was reading the image before passing it here

Reply:
You could instead make a small change to the function so that it will support both file and string inputs (i.e. a type check on the input). Then you can import it instead of copy pasting.